core/templates/pages/topic-viewer-page/topic-viewer-page.component.spec.ts

```diff
@@ -16,210 +16,226 @@
-  var ReadOnlyTopicObjectFactory = null;
-  var TopicViewerBackendApiService = null;
-  var UrlService = null;
-  var WindowDimensionsService = null;
-
-  var topicName = 'Topic Name';
-
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  beforeEach(angular.mock.inject(function($injector, $componentController) {
-    $q = $injector.get('$q');
-    var $rootScope = $injector.get('$rootScope');
-    AlertsService = $injector.get('AlertsService');
-    PageTitleService = $injector.get('PageTitleService');
-    ReadOnlyTopicObjectFactory = $injector.get('ReadOnlyTopicObjectFactory');
-    TopicViewerBackendApiService = $injector.get(
-      'TopicViewerBackendApiService');
-    UrlService = $injector.get('UrlService');
-    WindowDimensionsService = $injector.get('WindowDimensionsService');
-
-    $scope = $rootScope.$new();
-    ctrl = $componentController('topicViewerPage', {
-      $scope: $scope
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+
+import { ReadOnlyTopicObjectFactory } from
+  'domain/topic_viewer/read-only-topic-object.factory';
+import { TopicViewerBackendApiService } from
+  'domain/topic_viewer/topic-viewer-backend-api.service';
+import { TopicViewerPageComponent } from
+  'pages/topic-viewer-page/topic-viewer-page.component';
+import { AlertsService } from 'services/alerts.service';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowDimensionsService } from
+  'services/contextual/window-dimensions.service';
+import { PageTitleService } from 'services/page-title.service';
+
+
+describe('Topic viewer page', () => {
+  let httpTestingController = null;
+  let alertsService = null;
+  let pageTitleService = null;
+  let readOnlyTopicObjectFactory = null;
+  let topicViewerBackendApiService = null;
+  let urlService = null;
+  let windowDimensionsService = null;
+  let topicViewerPageComponent = null;
+
+  let topicName = 'Topic Name';
+  let topicUrlFragment = 'topic-frag';
+  let topicDict = {
+    topic_id: '1',
+    topic_name: 'Topic Name',
+    topic_description: 'Topic Description',
+    canonical_story_dicts: [{
+      id: '2',
+      title: 'Story Title',
+      node_titles: ['Node title 1', 'Node title 2'],
+      thumbnail_filename: '',
+      thumbnail_bg_color: '',
+      description: 'Story Description',
+      story_is_published: true
+    }],
+    additional_story_dicts: [],
+    uncategorized_skill_ids: [],
+    subtopics: [],
+    degrees_of_mastery: {},
+    skill_descriptions: {},
+    train_tab_should_be_displayed: true
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [TopicViewerPageComponent]
```

Comment:
You need to add components to declarations, not providers. Providers are for Dependency injection.
```suggestion
      declarations: [TopicViewerPageComponent]
```

Reply:
Done.